manual/src/tutorials/effectexamples.etex

```diff
@@ -0,0 +1,666 @@
+handles "E" and the outer handler handles "F". The call to "baz" returns
+"Hello, world!".
+
+\begin{caml_example}{verbatim}
+baz ()
+\end{caml_example}
+
+\subsection{s:effects-fibers}{Fibers}
+
+It is useful to know a little bit about the implementation of effect handlers
+to appreciate the design choices and their performance characteristics. Effect
+handlers are implemented with the help of runtime-managed, dynamically growing
+segments of stack called \textit{fibers}. The program stack in OCaml is a
+linked-list of such fibers.
+
+A new fiber is allocated for evaluating the computation enclosed by an effect
+handler. The fiber is freed when the computation returns to the caller either
+normally by returning a value or by raising an exception.
+
+At the point of "perform" in "foo" in the previous example, the program stack
+looks like this:
+
+\begin{caml}
+\begin{camlinput}
++-----+   +-----+   +-----+
+|     |   |     |   |     |
+| baz |<--| bar |<--| foo |
+|     |   |     |   |     |
+|     |   |     |   |     |
++-----+   +-----+   +-----+ <- stack_pointer
+\end{camlinput}
+\end{caml}
+
+The two links correspond to the two effect handlers in the program. When the
+effect "F" is handled in "baz", the program state looks as follows:
+
+\begin{caml}
+\begin{camlinput}
++-----+                   +-----+   +-----+
+|     |                   |     |   |     |   +-+
+| baz |                   | bar |<--| foo |<--|k|
+|     |                   |     |   |     |   +-+
++-----+ <- stack_pointer  +-----+   +-----+
+\end{camlinput}
+\end{caml}
+
+The delimited continuation "k" is an object on the heap that refers to the
+segment of the stack that corresponds to the suspended computation. Capturing a
+continuation does not involve copying stack frames. When the continuation is
+resumed, the stack is restored to the previous state by linking together the
+segment pointed to by "k" to the current stack. Since neither continuation
+capture nor resumption requires copying stack frames, suspending the execution
+using "perform" and resuming it using either "continue" or "discontinue" are
+fast.
+
+\subsection{s:effects-unhandled}{Unhandled effects}
+
+Unlike languages such as Eff and Koka, effect handlers in OCaml do not provide
+\textit{effect safety}; the compiler does not statically ensure that all the
+effects performed by the program is handled. If effects do not have a matching
+handler, then "Unhandled" exception is raised at the point of the corresponding
+"perform". For example, in the previous example, "bar" does not handle the
+effect "F". Hence, we will get "Unhandled" exception when we run "bar".
+
+\begin{caml_example}{verbatim}
+try bar () with Unhandled -> "Saw unhandled exception"
+\end{caml_example}
+
+\subsection{s:effects-linearity}{Linear continuations}
+
+The delimited continuations in OCaml must be used linearly -- every captured
+continuation must be resumed either with a "continue" or "discontinue" exactly
```

Comment:
Suggestion: "must be resumed exactly once, either by `continue` or by `discontinue`". This rule could appear in bold face. And you could indicate that this rule has already been stated earlier (when discussing resource leaks in 3.1).